### ISSUE TITLE
Default Value Option

### ISSUE THREAD

**Issue by dianacamacho** [LaunchPadLab/decanter#104]
## Feature Description
Not sure if this is already an option or not (couldn't find anything), but optionally tell a decanter to set a default value when incoming args do not include the expected input key (rather than requiring a value or ignoring the key completely)

## Suggested Solution
Allow adding `default_value` as input `options`

eg:
 ```
args = { a: 1, b: 2 }

Decanter
  input :a, :integer
  input :b, :integer
  input :c, :string, default_value: nil

New Result With Default:  { a: 1, b: 2, c: nil } 
Old Result: { a: 1, b: 2 } # c is missing :/
```

PR #103 

## Additional Context

Using decanter to parse API response and build POROs for the response that dynamically define object attributes based on the decanter input keys. If a response happens to omit a key because it is empty the `attr_accessor` will not be created because currently decanter skips over keys not included in the incoming args 

(@chawes13 basically an extension of your use of decanter in DSS on the external API responses, but for Centaman where the responses aren't always consistent and keys are often missing when blank)

**Comment by chawes13** [comment]
@dianacamacho Do you have an existing workaround for this? If you're using a similar ObjectWrapper, it looks like explicitly setting the `attr_accessor` when the class is instantiated will override custom methods. E.g.,
```rb
class Test
  def initialize(args={})
    self.class.__send__(:attr_accessor, 'hello')
    instance_variable_set("@hello", "it's me")
  end

  def hello
    'world'
  end
end

Test.new.hello
=> "it's me"
```

**Comment by chawes13** [comment]
@dianacamacho Couple of questions for you regarding how you envision this feature working for some edge cases.
- Would `default_value` be used in place of missing keys **and** `nil` values? E.g.,
```rb
input :a, :string
input :b, :string
input :c, :string, default_value: 'three'

{ a: 'one', b: 'two' } # => {  a: 'one', b: 'two', c: 'three }
{ a: 'one', b: 'two', c: nil } # => { a: 'one', b: 'two', c: nil } OR { a: 'one', b: 'two', c: 'three' }
```
- Would the value provided in `default_value` be parsed via the specified parser or taken "as-is? E.g.,
```rb
input :b, :string, default_value: 2
{ a: 'one' } # => { a: 'one', b: 2 } OR { a: 'one', b: '2' }
```

**Comment by dianacamacho** [comment]
@chawes13 i'm envisioning it as in place of missing keys and keeping explicit setting of attrs as nil as is. right now it's parsed as is (Mainly because i didn't have the time to figure out how to get the parser to apply, but i think that would be ideal)


Re: explicitly setting the attr_accessor, the point of this is avoiding having to configure attrs in multiple places, the decanter being the main area where you can see what attributes are being defined. in the centaman case, there is often several attributes that could or could not be missing from the response if they are empty, so i would be setting the attribute in the decanter as well as in the object class. ideally that only has to happen in one place. 

Re: method_missing, since this is meant to define attr_accessors, it feels weird for that to happen there. if i were to pull up an object in console and make a typo, i would assume that would then add that typo as an attr_accessor? which seems really strange

**Comment by chawes13** [comment]
I suppose methods aren't included in `as_json` either, which I believe was part of the original implementation. Point taken on defining things in multiple places

**Comment by chawes13** [comment]
Potentially we could add in a global and/or per decanter setting in a follow-up that would allow the user to specify whether `.blank?` values should be overridden by the supplied default (similar to `strict`)?

**Comment by dianacamacho** [comment]
@chawes13 yup, `as_json` is one of the things i would want to keep. 

follow-up idea makes sense 👍

**Comment by chawes13** [comment]
Resolved by #103
